src/pointer-analysis/value_set.cpp

```diff
@@ -674,9 +689,12 @@ void value_sett::get_value_set_rec(
             i->negate();
```

Comment:
This always negates the constant value `i`, but now we allow `i - ptr` and so the negation should respect the order of the operands. Also note that since the code has changed from explicitly checking for binary operands to n-ary, it is (in theory - I don't know that we construct n-ary pointer +/-) not sufficient to simply gather all the constant values, since we could have `i1 - ptr - i2`.

Reply:
Thank you for calling out these scenarios! None of them should be possible (a `minus_exprt` is a binary expression, and subtraction of a pointer requires should only be possible when the left-hand side also is a pointer), so I've added a `DATA_INVARIANT` to make sure we catch all those cases, should they unexpectedly occur.